azure/functions/decorators/function_app.py

```diff
@@ -302,22 +264,28 @@ def decorator():
 
         return wrap
 
-    def _add_http_app(self,
-                      http_middleware: Union[
-                          AsgiMiddleware, WsgiMiddleware]) -> None:
-        """Add a Wsgi or Asgi app integrated http function.
 
-        :param http_middleware: :class:`AsgiMiddleware` or
-        :class:`WsgiMiddleware` instance.
+class HttpFunctionsAuthLevelMixin(ABC):
```

Comment:
What is the Mixin at the end of the classname? Can we rename to IHttpFunctionsAuthLevel . Similar comment to the other interfaces. We can add an 'I' before the class name to differentiate interfaces from other classes

Reply:
Mixin is implying this is not is-a relationship, https://www.pythontutorial.net/python-oop/python-mixin/#:~:text=What%20is%20a%20mixin%20in%20Python%20A%20mixin,mixin%20bundles%20a%20set%20of%20methods%20for%20reuse.

For I prefix, I do not see any other abstract classes in library use it. Lets keep it consistent. Also in python standard libraries do not see a pattern of using I prefix
<img width="253" alt="image" src="https://user-images.githubusercontent.com/12449837/166721895-2c1a80b2-5255-45b1-aeb0-8d9e12a7fdb4.png">